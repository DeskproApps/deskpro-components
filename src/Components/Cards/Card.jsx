import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { objectKeyFilter } from '../../utils/objects';

/**
 * Renders a standard card.
 */
const Card = ({ className, children, ...props }) => (
  <div
    className={classNames('dp-card', className)}
    {...objectKeyFilter(props, Card.propTypes)}
  >
    {children}
  </div>
);

Card.propTypes = {
  /**
   * CSS classes to apply to the element.
   */
  className: PropTypes.string,
  /**
   * Children to render.
   */
  children:  PropTypes.node
};

Card.defaultProps = {
  className: '',
  children:  ''
};

export default Card;
