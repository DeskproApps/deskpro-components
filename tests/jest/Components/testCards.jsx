import React from 'react';
import renderer from 'react-test-renderer';
import { Card } from '../../../src/Components/Cards/index';

describe('>>> Buttons --- Cards', () => {
  it('+++capturing Snapshot of Card', () => {
    const renderedValue = renderer.create(<Card>
      <h1>Test content</h1>
    </Card>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
