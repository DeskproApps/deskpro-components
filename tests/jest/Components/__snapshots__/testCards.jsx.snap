// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Buttons --- Cards +++capturing Snapshot of Card 1`] = `
<div
  className="dp-card"
>
  <h1>
    Test content
  </h1>
</div>
`;
